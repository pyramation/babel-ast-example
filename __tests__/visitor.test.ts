import babelTraverse from '@babel/traverse';
import { parse, ParserPlugin } from '@babel/parser';
import generate from '@babel/generator';
import * as t from '@babel/types';

const expectCode = (ast) => {
    expect(
        generate(ast).code
    ).toMatchSnapshot();
}

const printCode = (ast) => {
    console.log(
        generate(ast).code
    );
}


it('works', () => {

    const code = `
export interface MyInterface {
    a: string;
    b: number;
}
`;

    const plugins: ParserPlugin[] = [
        'typescript',
    ];

    const ast = parse(code, {
        sourceType: 'module',
        plugins
    });

    babelTraverse(ast, {
        TSTypeAliasDeclaration(path) {
            console.log(path.node.id.name, path.parentPath.node);
        },
        TSInterfaceDeclaration(path) {
            path.node.id.name = 'MyOtherInterface'
            console.log(path.node.id.name, path.parentPath.node);
        }
    });


    expectCode(ast);


});