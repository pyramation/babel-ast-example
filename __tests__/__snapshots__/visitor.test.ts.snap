// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
"export interface MyOtherInterface {
  a: string;
  b: number;
}"
`;
